PSOperations/OperationQueue.swift

```diff
@@ -103,7 +100,14 @@ public class OperationQueue: NSOperationQueue {
         }
         
         delegate?.operationQueue?(self, willAddOperation: operation)
-        super.addOperation(operation)   
+        super.addOperation(operation)
+        
+        /*
+            Indicate to the operation that we've finished our extra work on it
+            and it's now it a state where it can proceed with evaluating conditions,
+            if appropriate.
+        */
+        psOperation?.didEnqueue()
```

Comment:
I think having another `if let op = operation as? Operation` down here is much more readable than saving a variable way up at the top which is only used in the last line of the method.

Also, I'm not fond of the variable name `psOperation`.